draft-ietf-quic-http.md

```diff
@@ -201,6 +201,21 @@ additional framing. Note that a request or response without a body will cause
 this stream to be half-closed in the corresponding direction without
 transferring data.
 
+Because the message control stream contains HPACK data which manipulates
+connection-level state, the message control stream MUST NOT be closed with a
+stream-level error.  If an implementation chooses to reject a request with a
+QUIC error code, it MUST trigger a RST_STREAM on the data stream only.  An
+implementation MAY close (FIN) a message control stream without completing a
+full HTTP message if the data stream has been abruptly closed.  Data on message
+control streams MUST be fully consumed, or the connection terminated.
+
+If a message control stream is terminated abruptly by the transport layer (e.g.
+with QUIC_TOO_MANY_OPEN_STREAMS), any frames already sent on that stream MUST
+be re-played on another message control stream as soon as the maximum number of
+streams permits.  If the implementation no longer requires the response to this
+request, it MAY abruptly terminate the corresponding data stream at the same
+time.
```

Comment:
I think the concern is that the transport layer doesn't necessarily know what's a data stream versus a control stream.  And if the application layer has said "Give me no more than 10 streams at a time," it seems... odd for the QUIC layer to then say "Here are 20 streams that have just arrived; if you'd like to shoot 10 of them, feel free."  The app already set the limit, and that limit should be enforced.

And if that enforcement causes the whole connection to get torn down, that might be okay.